extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -371,29 +372,39 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
 
 	protected function sendEmails( array $watchers, WallNotificationEntity $notification ) {
 		$text = strip_tags( $notification->data_noncached->msg_text, '<p><br>' );
-		$text = substr( $text, 0, 3000 ) . ( strlen( $text ) > 3000 ? '...' : '' );
-
-		$textNoHtml = preg_replace( '#<br\s*/?>#i', "\n", $text );
-		$textNoHtml = trim( preg_replace( '#</?p\s*/?>#i', "\n", $textNoHtml ) );
-		$textNoHtml = substr( $textNoHtml, 0, 3000) . ( strlen( $textNoHtml ) > 3000 ? '...' : '' );
+		$text = mb_substr( $text, 0, self::MAX_ABSTRACT_LENGTH, 'UTF-8' )
+			. ( mb_strlen( $text, 'UTF-8' ) > self::MAX_ABSTRACT_LENGTH ? '...' : '' );
+		$textNoHtml = trim( preg_replace( '#</?(p|br)[^/>]*/?>#i', "\n", $text ) );
 
 		$entityKey = $notification->getId();
 
-		if ( empty( $this->uniqueUsers[$entityKey] ) ){
+		if ( empty( $this->uniqueUsers[$entityKey] ) ) {
 			$this->uniqueUsers[$entityKey] = [];
 		}
 
-		foreach ( $watchers as $val ){
-			$watcher = $this->getUser( $val );
-			$mode = $watcher->getOption( 'enotifwallthread' );
-
-			if ( !empty( $mode ) && $watcher->getId() != 0 && (
```

Comment:
All the logic here got moved into `canSendToWatcher()`, but it doesn't look like it's being called any more. Should we add a call to `canSendToWatcher()` somewhere around here? Like above the call to `getEmailExtensionController()`?